### PR TITLE
update test and method to return null for prop type

### PR DESCRIPTION
**Pull request by ros3bud**
@cgrayson, LGTY? 

for property.type:
- add test to account for null and strange values
- update method to return null

(As a side note, there are several methods in this integration which could also use this treatment. However, the methods are for values that are not even currently being passed and apparently we don't even have solid docs for Costco. Do you think we should improve this integration further on an as-needed basis?)

### CODE REVIEW

File: spec/inbound_spec.coffee [hunk 1]
```diff
@@ -148,7 +148,7 @@ describe 'Inbound', ->
 
     describe 'Property', ->
       it 'should parse property type as single family', ->
-        assert.equal integration.request(req(body())).mortgage.property.type, 'single family'
+        assert.equal integration.request(req(body(prop_type: 'single'))).mortgage.property.type, 'single family'
```
Comment (cgrayson) [inline spec/inbound_spec.coffee]: Looks like you replaced 'single_family_residence' in this test with 'single', was that on purpose?

File: spec/inbound_spec.coffee [hunk 2]
```diff
@@ -168,6 +168,12 @@ describe 'Inbound', ->
       it 'should parse property type as multi family', ->
         assert.equal integration.request(req(body(prop_type: 'multi-family_dwelling_unit'))).mortgage.property.type, 'multi family'
 
+      it 'should parse property type as null', ->
+        assert.equal integration.request(req(body( prop_type: null ))).mortgage.property.type, null
```
Comment (cgrayson) [inline spec/inbound_spec.coffee]: The way the test `body` is created, this will make it be `…&property_type=null&…`, which probably isn't what will be sent by Costco. If they don't send a value for that, it will probably be empty (e.g., `…& property_type=&…`) or `property_type` will be missing altogether.

File: src/inbound.coffee [hunk 3]
```diff
@@ -249,6 +248,8 @@ parsePropertyType = (value) ->
     when /multi/.test(value) then 'multi family'
     when /mobile/.test(value) then 'mobile home'
     when /coop/.test(value) then 'coop'
+    else null
+
```
Comment (cgrayson) [inline src/inbound.coffee]: For the side note question on this PR, I'm not sure. I assume you're addressing #25, but I don't know the impetus behind that issue. If there are leads that we've seen with mishandled data, then of course we need to fix it. Otherwise, unless we're making other changes or fixes and tidying this up while we're in the code, I wouldn't think it's a priority.

File: spec/inbound_spec.coffee (hunk 1 above)
Comment (ros3bud) [inline spec/inbound_spec.coffee]: This is supposed to be this way in order to make sure that the regex /single/.test(value) translates to 'single family'.

File: spec/inbound_spec.coffee (hunk 2 above)
Comment (ros3bud) [inline spec/inbound_spec.coffee]: Updated line 171 in the spec to not pass anything.

File: src/inbound.coffee (hunk 3 above)
Comment (ros3bud) [inline src/inbound.coffee]: Thanks, yes, just making updates to tidy things up while there were other items that were necessary.

File: spec/inbound_spec.coffee (hunk 1 above)
Comment (cgrayson) [inline spec/inbound_spec.coffee]: The best test in an inbound integration is to use data as the vendor sends it. The previous code tested for "single_family_residence", and now this is testing for "single". Was that change made because of a better understanding of what Costco actually sends in this field? 

(Note that `/single/.test(value)` is true both when `value` is "single" and when it's "single_family_residence".)

**Comment by ros3bud** [comment]
@cgrayson, updated.

**Comment by cgrayson** [comment]
LGTM
